CMakeLists.txt

```diff
@@ -1,7 +1,21 @@
 cmake_minimum_required(VERSION 3.5)
 project(raspicam2)
 
-set(CMAKE_CXX_STANDARD 14)
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
```

Comment:
Why do we first have to check `CMAKE_CXX_STANDARD`? Can we not just set it directly without checking first? I think this way someone could set a different standard that is not compatible with at least C++14, which would result in compilation errors.

Reply:
Again, this is the customary configuration in ROS2: https://index.ros.org/doc/ros2/Tutorials/Ament-CMake-Documentation/#id6
It appears in virtually all the documentation, code samples and courses at least from Eloquent (when I switched to ROS2)